equinix/resource_fabric_network.go

```diff
@@ -0,0 +1,430 @@
+				Schema: equinix_fabric_schema.ChangeLogSch(),
+			},
+		},
+		"connections_count": {
+			Type:        schema.TypeInt,
+			Computed:    true,
+			Description: "Number of connections associated with this Access point",
+		},
+	}
+}
+func resourceFabricNetwork() *schema.Resource {
+	return &schema.Resource{
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(6 * time.Minute),
+			Update: schema.DefaultTimeout(10 * time.Minute),
+			Delete: schema.DefaultTimeout(6 * time.Minute),
+			Read:   schema.DefaultTimeout(6 * time.Minute),
+		},
+		ReadContext:   resourceFabricNetworkRead,
+		CreateContext: resourceFabricNetworkCreate,
+		UpdateContext: resourceFabricNetworkUpdate,
+		DeleteContext: resourceFabricNetworkDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+		Schema: FabricNetworkResourceSchema(),
+
+		Description: "Fabric V4 API compatible resource allows creation and management of Equinix Fabric Network",
+	}
+}
+
+func resourceFabricNetworkCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client := meta.(*config.Config).FabricClient
+	ctx = context.WithValue(ctx, v4.ContextAccessToken, meta.(*config.Config).FabricAuthToken)
+	schemaNotifications := d.Get("notifications").([]interface{})
+	notifications := equinix_fabric_schema.NotificationsToFabric(schemaNotifications)
+	schemaLocation := d.Get("location").(*schema.Set).List()
+	location := equinix_fabric_schema.LocationToFabric(schemaLocation)
+	project := v4.Project{}
+	schemaProject := d.Get("project").(*schema.Set).List()
+	if len(schemaProject) != 0 {
+		project = equinix_fabric_schema.ProjectToFabric(schemaProject)
+	}
+	netType := v4.NetworkType(d.Get("type").(string))
+	netScope := v4.NetworkScope(d.Get("scope").(string))
+
+	createRequest := v4.NetworkPostRequest{
+		Name:          d.Get("name").(string),
+		Type_:         &netType,
+		Scope:         &netScope,
+		Location:      &location,
+		Notifications: notifications,
+		Project:       &project,
+	}
+
+	fabricNetwork, _, err := client.NetworksApi.CreateNetwork(ctx, createRequest)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	d.SetId(fabricNetwork.Uuid)
+
+	if _, err = waitUntilFabricNetworkIsProvisioned(d.Id(), meta, ctx); err != nil {
+		return diag.Errorf("error waiting for Network (%s) to be created: %s", d.Id(), err)
+	}
+
+	return resourceFabricNetworkRead(ctx, d, meta)
+}
+
+func resourceFabricNetworkRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client := meta.(*config.Config).FabricClient
+	ctx = context.WithValue(ctx, v4.ContextAccessToken, meta.(*config.Config).FabricAuthToken)
+	fabricNetwork, _, err := client.NetworksApi.GetNetworkByUuid(ctx, d.Id())
+	if err != nil {
+		log.Printf("[WARN] Fabric Network %s not found , error %s", d.Id(), err)
+		if !strings.Contains(err.Error(), "500") {
+			d.SetId("")
+		}
+		return diag.FromErr(err)
```

Comment:
Please update all API error responses with the guidance listed in earlier comment.